VirtoCommerce.Storefront/Routing/SlugRouteService.cs

```diff
@@ -201,10 +201,7 @@ protected virtual async Task LoadObjectsAndBuildFullSeoPaths(string objectType,
             {
                 foreach (var seo in objects)
                 {
-                    if (seoPaths.ContainsKey(seo.ObjectId))
-                    {
-                        seo.SeoPath = seoPaths[seo.ObjectId];
-                    }
+                    seo.SeoPath = seoPaths[seo.ObjectId] ?? seo.SeoPath;
```

Comment:
This is bad practice. If you don't add `.WithDefaultValue(null)` you'll get an exception.
Create an extension method wich will check if the key exists and use it like this:
`seoPaths.GetValueSafe(seo.ObjectId, null) ?? seo.SeoPath;`
``` csharp
    public static class DictionaryExtensions
    {
        public static T GetValueSafe<T>(this IDictionary<string, T> dictionary, string key, T defaultValue)
        {
            T result;

            if (key != null && dictionary.TryGetValue(key, out var value))
            {
                result = value;
            }
            else
            {
                result = defaultValue;
            }

            return result;
        }

        public static bool TryGetValueSafe<T>(this IDictionary<string, T> dictionary, string key, out T value)
        {
            bool result;

            if (key != null)
            {
                result = dictionary.TryGetValue(key, out value);
            }
            else
            {
                result = false;
                value = default(T);
            }

            return result;
        }
    }
```